inc/editor/blocks/mailchimp-subscribe.php

```diff
@@ -1,104 +1,81 @@
 <?php
 /**
- * Define the behavior of the Mailchimp Subscribe block.
+ * Modify the behavior of the Mailchimp Subscribe block.
  */
 
 namespace WMF\Editor\Blocks\MailChimpSubscribe;
 
-use WMF\Customizer\Connect;
-
 /**
- * Bootstrap hooks for the mailchimp block
+ * Bootstrap hooks for modifying the mailchimp plugin block.
  */
 function bootstrap() {
-	add_action( 'init', __NAMESPACE__ . '\\register' );
+	add_action( 'wp_enqueue_scripts', __NAMESPACE__ . '\\mailchimp_resources_dequeue', 11 );
+	add_filter( 'render_block', __NAMESPACE__ . '\\filter_mailchimp_output', 10, 2 );
+
+	// Remove Mailchimp styles, we'll provide our own.
+	update_option( 'mc_nuke_all_styles', true );
 }
 
 /**
- * Render mailchimp subscribe block. This is required because wp_kses_post will
- * strip <form> and <input> tags. For the mailchimp subscribe we need those
- * tags. Other than these changes, this is de facto a static block.
- *
- * @param array  $block_attributes Saved block attributes.
- * @param string $content          Rendered block inner content.
+ * Remove uneeded Mailchimp styles & JS that are loaded by the Mailchimp plugin.
  */
-function render_block( $block_attributes, $content ) {
-	$action            = get_theme_mod( 'wmf_subscribe_action', Connect::defaults( 'wmf_subscribe_action' ) );
-	$additional_fields = get_theme_mod( 'wmf_subscribe_additional_fields',
-	Connect::defaults( 'wmf_subscribe_additional_fields' ) );
-	$additional_fields = kses_input_fields( $additional_fields );
-
-	/*
-	 * This setting was misused for actual content, strip the current content on
-	 * production. **Can be removed once the setting is correctly used with only
-	 * fields**
-	 */
-	$additional_fields = str_replace( "This mailing list is powered by MailChimp. The Wikimedia Foundation will handle your personal information in accordance with this site's privacy policy.", '', $additional_fields );
-
-	$input_placeholder = empty( $block_attributes['inputPlaceholder'] ) ?
-		__( 'Email address', 'shiro' ) :
-		$block_attributes['inputPlaceholder'];
+function mailchimp_resources_dequeue() {
+	// Dequeue resources that aren't needed for the email address field, like datepicker.
+	wp_dequeue_style( 'flick' );
+	wp_dequeue_script( 'jquery-ui-datepicker' );
+	remove_action( 'wp_head', 'mc_datepicker_load' );
 
-	$form_start = '<form action="' . esc_url( $action ) . '" method="POST" class="mailchimp-subscribe__form">';
-	$form_end   = '</form>';
-	$input_field = '<input' .
-		' class="mailchimp-subscribe__input-field"' .
-		' id="wmf-subscribe-input-email"' .
-		' name="EMAIL"' .
-		' placeholder="' . esc_attr( $input_placeholder ) . '"' .
-		' required=""' .
-		' type="email" />';
+	// Get blocks and dequeue all other mailchimp resources if the block isn't in use.
+	if ( has_block( 'mailchimp/mailchimp' ) ) {
```

Comment:
Is this block always going to be in post content? I'm wondering if the block might be included in a template part or something, and if in that case `has_block` wouldn't return true.